gcomm/src/gcomm/view.hpp

```diff
@@ -226,15 +227,16 @@ namespace gcomm
     class ViewState
     {
     public:
-        ViewState(UUID& my_uuid, View& view):
+        ViewState(UUID& my_uuid, View& view, gu::Config& conf):
                 my_uuid_(my_uuid),
-                view_(view)
+                view_(view),
+                file_name_(get_viewstate_file_name(conf))
         { }
         std::ostream& write_stream(std::ostream& os) const;
         std::istream& read_stream(std::istream& is);
-        void write_file(const char* fname = NULL) const;
-        bool read_file(const char* fname = NULL);
-        static void remove_file(const char* fname = NULL);
+        void write_file() const;
+        bool read_file();
+        static void remove_file(gu::Config& conf);
```

Comment:
Yes, in most cases it is used when there is no ViewState object in hands:
- in gcomm::gmcast::Proto::handle_failed, there we do not have neithe ViewState, nor conf. I had to add accessor to gmcast.hpp.
- in gcomm::PC::close - we do not have ViewState object. 
The only exception is gcomm::PC::PC, there we do have ViewState object.

There is also unit test use of remove_file, but i do not count it.